### ISSUE TITLE
pre-commit hook defeats path-limited commits by staging issues.jsonl

### ISSUE THREAD

**Issue by verioussmith** [Dicklesworthstone/beads_rust#482]
## Defect

The generated Beads pre-commit hook can inject `.beads/issues.jsonl` into a path-limited Git commit.

Observed with `br 0.2.16`:

```bash
git commit --only AGENTS.md -m "docs: ..."
git diff-tree --no-commit-id --name-status -r HEAD
```

Expected:

```text
M AGENTS.md
```

Actual:

```text
M .beads/issues.jsonl
M AGENTS.md
```

No Beads payload is included here; only the unexpected path is reported.

## Root cause

The installed hook runs `br sync --flush-only`, then unconditionally runs:

```sh
git add "$BEADS_DIR/issues.jsonl"
```

for a regular worktree. During `git commit --only <path>`, this mutates the commit's effective/temporary index and defeats the caller's explicit path restriction.

## Impact

- Unrelated task-ledger state can enter carefully scoped commits.
- Sensitive/local-only Beads data can contaminate a commit intended for portable publication.
- `git commit --only` / pathspec isolation becomes untrustworthy.

## Fix plan

1. Add a regression fixture with a dirty `.beads/issues.jsonl` and unrelated file.
2. Cover `git commit --only FILE`, `git commit FILE`, ordinary full commit, `-a`, and worktree cases.
3. Never stage Beads state into a path-limited/temporary commit index. Detect that mode safely, or stop mutating the index in pre-commit and fail with an explicit staging instruction when ledger inclusion is required.
4. Preserve flush durability without widening the caller's commit pathspec.
5. Document hook behavior and expose a typed diagnostic when automatic staging is intentionally skipped.

## Acceptance

- A commit invoked with `--only AGENTS.md` contains exactly `AGENTS.md`.
- The pre-existing index and working tree retain unrelated Beads changes byte-for-byte.
- Ordinary intended Beads commits still flush/include ledger state through the documented path.
- Regression tests pass for normal repositories and linked worktrees.

**Comment by Dicklesworthstone** [comment]
Thanks for the precise write-up; the mechanism is exactly as you describe, but the hook is not something `br` generates or installs.

What I verified:

- `br` has never installed, generated, or run git hooks (README "Limitations" table, `docs/SYNC_SAFETY.md` "What br sync will never do"). There is no hook template anywhere in this repository, and no code path emits `git add`. The `bd hooks install` command and its generated pre-commit script belong to the Go `beads` tool; `br 0.2.16` is also an old line (current release is 0.5.7).
- Reproduced your observation with a hand-written hook in a scratch repo on git 2.50: during `git commit --only <path>` and `git commit <path>`, git builds the commit from a temporary index and exports it as `GIT_INDEX_FILE=.git/next-index-<pid>.lock` (in a linked worktree: `.git/worktrees/<name>/next-index-<pid>.lock`). A `git add` inside the hook writes into that temporary index and widens the commit. For `-a` / `--include` the variable is `.git/index.lock`, and for a plain `git commit` it is the real index.

Since the fix belongs in whichever tool generated your hook, I did not add hook installation to `br`. What I did (2f9518a2 (docs-only; the change itself is in 056d271d)): `docs/SYNC_SAFETY.md` now carries a reference hook that keeps the flush and skips the staging step in path-limited mode, verified against `--only`, implicit pathspec, staged-only, `-a`, and a linked worktree:

```sh
#!/bin/sh
br sync --flush-only --quiet || exit 1
case "$(basename "${GIT_INDEX_FILE:-index}")" in
  next-index-*) exit 0 ;;   # path-limited commit: do not widen it
esac
git add .beads/issues.jsonl
```

With that guard, `git commit --only AGENTS.md` contains exactly `AGENTS.md`, the working tree and index keep the unrelated `.beads` change untouched, and ordinary commits still include the ledger. If the hook came from `bd hooks install`, please file the same finding against the Go beads repository; it is a one-line change there. Closing here as not a `br` defect.
